### ISSUE TITLE
Fine-tuning the model on the EchoNet Dataset

### ISSUE THREAD

**Issue by perecanals** [berkindurmus/SLIViT#19]
Hi!

First of all, congrats on the great work and thank you for open-sourcing it!

I have been trying to execute the training on the EchoNet US dataset. I downloaded the dataset an modified the paths on the csv file, but then I noticed that a preprocessing step is needed to pass the videos in `.avi` format to  `.tiff`. If I infer correctly, I understand that preprocessing should turn the `.avi` videos into a set of 32 `.tiff` files (in the paper you mention that "To standardize the data we followed the same approach that the EchoNet paper authors took and sampled 32 equally-spaced frames per volume."). Could you share the code for the SLIViT preprocessing for the EchoNet dataset? That would be super helpful.

If that is not possible, perhaps you can answer the following question: is the preprocessing limited to the cropping to 112x112 pixels, 32 equally-spaced frames and 12px padding on all sides (gathered from the EchoNet paper) or did you apply any sort of intensity adjustment (e.g., normalization, contrast adjustment...).

Also, I noticed that in the suggested terminal snippet that you provide for fine-tuning in the US dataset (`python slivit_train.py --dataset3d ultrasound --meta_csv ./Dsets/ultrasound_meta.csv --bbpath ./Checkpoints/convnext_bb_kermany.pth --nObb_feat 4 --nslc 32 --depth 5 --dim 256 --heads 32`) you use the `convnext_bb_kermany` checkpoint. Does this mean that the US fine-tuning is done after pre-training the dataset on the OCT dataset? In this case, is the `slivit_ultrasound.pth` checkpoint the model trained with the provided snippet? Or is this a typo and it should be the `slivit_ultrasound.pth` checkpoint?

Thank you!

**Comment by berkindurmus** [comment]
The preprocessing involves simply rescaling the images to 224x224 and sampling 32 slices with equal spacing. Regarding the backbone, it is pretrained on 2D images, and there are several different pretrained backbones available, each based on different imaging modalities. Details about these backbones can be found in the paper. After the backbone is pretrained, the entire model is fine-tuned using 3D ultrasound (US) videos. You can also check the main repo for the more detailed guidance https://github.com/cozygene/SLIViT

**Comment by perecanals** [comment]
Oops sorry, I think I got the link to this repo from the preprint and did not realise that the one in the final paper was a different one! I'll check out the main repo, thank you @berkindurmus !
